### ISSUE TITLE
400 response during AddOrUpdateReport (for some users)

### ISSUE THREAD

**Issue by brett-lynnes** [alexa/avs-device-sdk#2043]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

## Briefly summarize your issue:
While logging into AVS on our device for the first time, some users (using the iOS companion app, not the Android companion app) are getting rejected during Discovery. The error from the server is a 400 BAD_CLIENT_REQUEST but the request appears to be valid. Also most of our users are able to log in appropriately, so this appears to be some server based problem.

## What is the expected behavior?
All users should be able to use their device after authorizing it. Attached is the DEBUG9 logs from a device that is unable to get through AddOrUpdateReport. Also attached is the json that is being sent in the body of that request from the device that is failing (8f3c) and one from a device that is succeeding (abad). You can diff them if you like, but I was unable to notice any substantial differences (the endpoints are reordered).

## What behavior are you observing?
~5% of users that used our iOS companion app (0% of Android users had this issue) are able to successfully authorize and get a token/refresh token, but when our device sends the AddOrUpdateReport during Discovery the server responds with an error 400. Logging out of Amazon and back in again on our device does not fix this issue. The only workaround we know is to have the user log out of the Amazon Shopping app on their iOS phone. Then log out of Amazon on our device and re-log into Amazon using our iOS companion app while Amazon Shopping app is closed.

[8f3cCAP.log](https://github.com/alexa/avs-device-sdk/files/9050558/8f3cCAP.log)
[8f3cCAP.txt](https://github.com/alexa/avs-device-sdk/files/9050559/8f3cCAP.txt)
[abadCAP.txt](https://github.com/alexa/avs-device-sdk/files/9050560/abadCAP.txt)

## Provide the steps to reproduce the issue, if applicable:
We have been unable to replicate the problem reliably. Users who encounter this have many different versions of iOS phones. We theorize it has something to do with cached credentials from Amazon Shopping or some missing scope that is automatically pulled when you are already signed in.

Basically: we theorize the issue is actually in the LWA portion of the code (that is succeeding), and probably on the iOS companion app code (or rather the Amazon libraries that the companion app relies on), but is only 
manifesting when the device attempts to do Discovery.

### What version of the AVS Device SDK are you using?
      <1.26.0>

### Tell us what hardware you're using:
- [x] Desktop / Laptop
- [ ] Raspberry Pi
- [ ] Other - tell us more:

### Tell us about your OS (Type & version):
- [x] Linux
- [ ] MacOS 
- [ ] Raspbian Stretch
- [ ] Raspbian Jessy
- [ ] Other - tell us more:

**Comment by kjkh** [comment]
Hi @brett-lynnes,

Sorry for the delayed response. We are contacting the appropriate internal teams to investigate further. To help aid with this, would you be able to reproduce the issue and attach fresh logs?

**Comment by blazusiak-amazon** [comment]
Hello @brett-lynnes,
Are you still experiencing this? Would you be able to reproduce this and attach fresh logs?

**Comment by kclchan** [comment]
I am closing this issue due to inactivity.  If you have further questions, please open a new ticket and refer to this closed one.
